docs/MAINTAINING.md

```diff
@@ -67,6 +68,24 @@ Output from acceptance testing in AWS GovCloud (US):
 
 #### Dependency Updates
 
+##### Go Default Version Update
+
+This project typically upgrades its Go version for development and testing shortly after release to get the latest and greatest Go functionality. Before beginning the update process, ensure that you review the new version release notes to look for any areas of possible friction when updating.
+
+Create an issue to cover the update noting down any areas of particular interest or friction.
+
+Ensure that the following steps are tracked within the issue and completed within the resulting pull request.
+
+- Update go version in `go.mod`
+- Verify all formatting, linting, and testing works as expected
+- Verify `gox` builds for all currently supported architectures
```

Comment:
Here's the verification command we use today, which matches the TeamCity release scripts:

```
gox -os='linux darwin windows freebsd openbsd solaris' -arch='386 amd64 arm' -osarch='!darwin/arm !darwin/386' -ldflags '-s -w -X aws/version.ProviderVersion=99.99.99 -X aws/version.ProtocolVersion=4' -output 'results/{{.OS}}_{{.Arch}}/terraform-provider-aws_v99.99.99_x4' .
```

In the (probably near) future, this will likely be rolled into a `goreleaser` configuration and automatically tested in PRs instead. 👍